### PR TITLE
Add Business Insider

### PR DESCRIPTION
**Pull request by MaxDall**
Adds [Business Insider](https://www.businessinsider.de/) as a publisher.

### CODE REVIEW

File: src/fundus/publishers/de/business_insider.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class BusinessInsiderParser(ParserProxy):
+    class V1(BaseParser):
+        _summary_selector = CSSSelector("article div.bi-bulletpoints > p")
+        _subheadline_selector = CSSSelector("article h2")
+
+        # The mark is to remove prepended text about machine translation
+        _paragraph_selector = XPath(
+            "//article //div[contains(@class, 'article-body')] "
+            "/p[not(mark[@class='has-inline-color has-cyan-bluish-gray-color'])]"
+        )
```
[the remark this answers is not in the document]
Comment (addie9800) [reply src/fundus/publishers/de/business_insider.py:25]: Some articles are incomplete due to the known issue with the paywall (e.g. https://www.businessinsider.de/wirtschaft/finanzen/ich-bin-29-jahre-alt-und-habe-6-immobilien-gekauft-das-sind-meine-5-tipps-wie-auch-ihr-das-schaffen-koennt/) not sure what can be done about it though

File: src/fundus/publishers/de/business_insider.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        def publishing_date(self) -> Optional[datetime.datetime]:
+            return generic_date_parsing(self.precomputed.ld.bf_search("datePublished"))
+
+        @attribute
+        def title(self) -> Optional[str]:
+            return self.precomputed.ld.bf_search("headline")
+
+        @attribute
+        def topics(self) -> List[str]:
+            return generic_topic_parsing(self.precomputed.meta.get("keywords"))
```
[the remark this answers is not in the document]
Comment (addie9800) [reply src/fundus/publishers/de/business_insider.py:50]: It seems to just occur within the insider-picks section of articles (another example https://www.businessinsider.de/insider-picks/technik/smartphones-unter-200-euro-empfehlungen/)

**Review by addie9800 — CHANGES_REQUESTED** [review]
Thanks for adding this

**Comment by MaxDall** [comment]
Oh wow, awesome 👍 Thanks for this really in depth review. Especially the link's to all the edge cases helping a lot. How did you find them?

**Comment by addie9800** [comment]
No problem. I crawled 50 articles and printed the title, authors, plaintext, url and topics. I then skimmed over the output and then I can check out the articles everytime something seems weird.

File: src/fundus/publishers/de/__init__.py [hunk 3]
```diff
@@ -190,3 +191,13 @@ class DE(PublisherEnum):
         sources=[NewsMap("https://www.waz.de/sitemaps/news.xml")],
         parser=WAZParser,
     )
+
+    BusinessInsider = PublisherSpec(
+        name="Business Insider",
+        domain="https://www.businessinsider.de/",
+        sources=[
```
[the remark this answers is not in the document]
Comment (MaxDall) [reply src/fundus/publishers/de/__init__.py:198]: Normally `NewsMap` and `Sitemap` are enough.

File: src/fundus/publishers/de/business_insider.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxDall) [reply src/fundus/publishers/de/business_insider.py:25]: > You could add or @class='has-text-align-right'

Good idea!

> Some articles are incomplete due to the known issue with the paywall

Sadly not that much. But judging by the article that's a cookie and not a paywall problem. I see what i can do.

File: src/fundus/publishers/de/business_insider.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        @attribute
+        def authors(self) -> List[str]:
+            return generic_author_parsing(self.precomputed.ld.bf_search("author"))
+
+        @attribute
+        def publishing_date(self) -> Optional[datetime.datetime]:
+            return generic_date_parsing(self.precomputed.ld.bf_search("datePublished"))
+
+        @attribute
+        def title(self) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (MaxDall) [reply src/fundus/publishers/de/business_insider.py:45]: I would stick to the rest of the parser and go with whatever is included in the ld.

File: src/fundus/publishers/de/business_insider.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaxDall) [reply src/fundus/publishers/de/business_insider.py:50]: I added some code to handle with this edge case

**Review by addie9800 — APPROVED** [review]
Looks good 👍

**Pull request merged**